uniffi_macros/src/util.rs

```diff
@@ -221,3 +222,72 @@ pub fn either_attribute_arg<T: ToTokens>(a: Option<T>, b: Option<T>) -> syn::Res
         }
     }
 }
+
+pub(crate) struct FfiConverterTagHandler {
+    tag: Option<Path>,
+}
+
+impl FfiConverterTagHandler {
+    pub(crate) fn generic_impl() -> Self {
+        Self { tag: None }
+    }
+
+    pub(crate) fn into_impl_and_tag_path(
+        self,
+        trait_name: &str,
+        ident: &Ident,
+    ) -> (TokenStream, Path) {
+        let trait_name = Ident::new(trait_name, Span::call_site());
+        match self.tag {
+            Some(tag) => (quote! { impl ::uniffi::#trait_name<#tag> for #ident }, tag),
+            None => (
+                quote! { impl<T> ::uniffi::#trait_name<T> for #ident },
+                Ident::new("T", Span::call_site()).into(),
+            ),
+        }
+    }
+}
+
+impl TryFrom<AttributeArgs> for FfiConverterTagHandler {
```

Comment:
Why use syn's `Meta` types instead of parsing the attributes using `syn::parse::Parse` like elsewhere? I already added a bunch of helpers to make that easier.

Reply:
I guess this is also the reason you didn't do `tag = crate::Tag` syntax, because `syn::Meta` doesn't support that?

Reply:
I tried, but couldn't figure out a good way to get from a `TokenStream` to an `[Attribute]` slicte to use with `parse_uniffi_attributes()`.  Maybe we could handle this as a follow-up.

Reply:
Yeah, I think I can take care of that next week.